src/agent/src/rpc.rs

```diff
@@ -2101,18 +2120,34 @@ async fn cdh_handler(oci: &mut Spec) -> Result<()> {
         .ok_or_else(|| anyhow!("Spec didn't contain mounts field"))?;
 
     for m in mounts.iter_mut() {
-        if m.destination().starts_with("/sealed") {
-            info!(
-                sl(),
-                "sealed mount destination: {:?} source: {:?}",
-                m.destination(),
-                m.source()
-            );
-            if let Some(source_str) = m.source().as_ref().and_then(|p| p.to_str()) {
-                cdh::unseal_file(source_str).await?;
-            } else {
-                warn!(sl(), "Failed to unseal: Mount source is None or invalid");
+        if let Some(source_path) = m.source().as_ref().and_then(|p| p.to_str()) {
+            // Check if source_path starts with "/run/kata-containers/shared/containers"
+            // For a volume mount path /mydir,
+            // the secret file path will be like this under the /run/kata-containers/shared/containers dir
+            // a128482812bad768f404e063f225decd425fc94a673aec4add45a9caa1122ccb-75490e32e51da3ff-mydir
+            // We can ignore few paths like: resolv.conf, termination-log, hostname,hosts,serviceaccount
+            if should_process_source_path(source_path) {
+                info!(
+                    sl(),
+                    "source file: {:?} destination: {:?}",
+                    source_path,
+                    m.destination()
+                );
+                // Call unseal_file. This function checks the files under the source_path
+                // for the sealed secret header and unseal it if the header is present.
+                // This is suboptimal as we are going through every file under the source_path.
+                // But currently there is no quick way to determine which volume-mount is referring
+                // to a sealed secret without reading the file.
+                // And relying on file naming heuristic is inflexible. So we are going with this approach.
+                if let Err(e) = cdh::unseal_file(source_path).await {
```

Comment:
Does this unseal the file in place? Is that dangerous in a shared dir?

Reply:
I mainly want to be sure that it's not exposed to the host. If we're unsealing something in the shared dir, that seems a bit dangerous but I forget exactly how it works.

Reply:
I'm assuming by host you mean worker node. It's not unsealed in the worker node. The configmaps and secrets are copied from worker node to the Kata VM, and inside the kata VM it's unsealed.

Reply:
I guess that hinges on using the copyfile thing to setup the kata-shared dir. That's probably fine.

Reply:
Right. we use copyFile to copy the sealed file and then unseal it inside the VM.